apiserver/facades/client/application/application.go

```diff
@@ -193,25 +186,26 @@ func newFacadeBase(stdCtx context.Context, ctx facade.ModelContext) (*APIBase, e
 
 	return NewAPIBase(
 		state,
-		domainServices.ExternalController(),
-		domainServices.Network(),
+		Services{
```

Comment:
I know this is in flight, and you've got a TODO to poke a bunch of discrete calls behind the service boundary, but into the future, this should suggest to us that we haven't quite our domain responsibilities right.

This facade could be the one with the most breadth though, hence the use of "suggest".

Reply:
This is 100% a valid point.

Let's break down to see what might be an issue and what we can eliminate.

```
ExternalControllerService: domainServices.ExternalController(),
NetworkService:            domainServices.Network(),
ModelConfigService:        domainServices.Config(),
CloudService:              domainServices.Cloud(),
CredentialService:         domainServices.Credential(),
MachineService:            domainServices.Machine(),
ApplicationService:        domainServices.Application(),
PortService:               domainServices.Port(),
StorageService:            domainServices.Storage(),
StubService:               domainServices.Stub(),
```

We know that stub is a temporary Service, so that will disappear.

Both cloud and credential look suspect. They generally tell me that something wants an environment. I think in both cases, they can go.

So that leaves us, with the following:

```
ExternalControllerService: domainServices.ExternalController(),
NetworkService:            domainServices.Network(),
ModelConfigService:        domainServices.Config(),
MachineService:            domainServices.Machine(),
ApplicationService:        domainServices.Application(),
PortService:               domainServices.Port(),
StorageService:            domainServices.Storage(),
```

This feels better but is still quite large.

The external controller service is the one that seems weird here, so I suspect we've not got this quite right.